velox/exec/TraceReader.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/file/FileSystems.h"
+#include "velox/exec/TraceInputStream.h"
+#include "velox/serializers/PrestoSerializer.h"
+#include "velox/vector/DecodedVector.h"
+#include "velox/vector/VectorStream.h"
+
+namespace facebook::velox::exec {
+class TraceReader {
+ public:
+  TraceReader(const std::string& path, const RowTypePtr& type)
+      : path_(path),
+        type_(type),
+        pool_(memory::MemoryManager::getInstance()
```

Comment:
Let's have a static trace memory pool in MemoryManager like spill memory pool? Thanks!